core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js

```diff
@@ -0,0 +1,35 @@
+(() => {
+	var redirectForm = document.getElementById('redirect-error');
+	if (!redirectForm) {
+		console.warn('This script expects to have an element with id="redirect-error" to be working.');
+	} else {
+		var urlToTest = redirectForm.getAttribute('data-url');
+		new Ajax.Request(urlToTest, {
+			onComplete : function(response) {
+				if (response.status !== 200) {
+					var context = redirectForm.getAttribute('data-context');
+					// to cover the case where the JenkinsRootUrl is configured without the context
+					if (context) {
+						new Ajax.Request(urlToTest, {
+							parameters: { testWithContext:true },
+							onComplete: function (response) {
+								if (response.status === 200) {
+									// this means the root URL was configured but without the contextPath, 
+									// so different message to display
+									redirectForm.style.display = "block";
+									redirectForm.querySelectorAll('.context-message').forEach(node => node.style.display = "block");
```

Comment:
I would suggest renaming the `.context-message` class to `.js-context-message`, to show is intent to be used by JS code and avoid a regression when refactoring CSS.

Reply:
Looks like I failed to pick an actual line :|

Reply:
It's a really good idea to js-xxx for stuff only used as tag for JavaScript without any CSS implication. Do we have a guideline for others?